lib/dynamoid/adapter_plugin/aws_sdk_v2.rb

```diff
@@ -539,12 +546,20 @@ def scan(table_name, scan_hash, select_opts = {})
 
         Enumerator.new do |y|
           # Batch loop, pulls multiple requests until done using the start_key
+          # by paging in by batch size rather than limit size
+          eval_count = 0
           loop do
             results = client.scan(request)
-
             results.data[:items].each { |row| y << result_item_to_hash(row) }
 
-            if((lk = results[:last_evaluated_key]) && batch)
+            # If limit is set then keep track and stop requesting once
+            # completed. This is eval_limit so we compare to scanned_count
+            # rather than resulting data
+            eval_count += results.scanned_count
```

Comment:
Looks like `scanned_count` returns count of loaded items before filtering. So if we provide some conditions with `where` clause `eval_limit` may return items less that was requested.

Should we use actual data size here?